In [73]:
import matplotlib.pyplot as plt
from matplotlib.colors import to_rgba

import seaborn as sns
import pandas as pd
import plotly.express as px
import scipy.stats as stats
import numpy as np

sns.set_theme(style="whitegrid")
sns.set_context('paper')

pd.set_option('display.max_rows', 500)

In [74]:
temperature = 65
data_col = 'execution_time'

In [75]:
df1 = pd.read_csv(f'../data/results_{temperature}_1.csv')
df2 = pd.read_csv(f'../data/results_{temperature}_2.csv')

df1['total_energy'] = df1['pkg_energy'] + df1['dram_energy']
df2['total_energy'] = df2['pkg_energy'] + df2['dram_energy']

progs = df1['program'].unique()
flags = df1['flag'].unique()

In [76]:
increase_df = []
for flag in flags:
    for prog in progs:
        base_mean = df1[(df1['flag'] == flag) & (df1['program'] == prog)][data_col].mean()
        diff_mean = df2[(df2['flag'] == flag) & (df2['program'] == prog)][data_col].mean()
        
        increase = ((diff_mean - base_mean) / base_mean) * 100
        
        increase_df.append([prog, flag, base_mean, diff_mean, increase])
increase_df = pd.DataFrame(increase_df, columns=['program', 'flag', 'base_mean', 'diff_mean', 'increase'])

In [77]:
stats.variation(increase_df['increase'])

-5.800776146403544

#### Variance

| Temperature | Energy Inc.        | Time Inc.          |
|:-----------:|-------------------:|-------------------:|
| 45          | 1.3090164987239208 | 0.7132850485966169 | 
| 55          | 37.84628463107378  | 20.92455721942198  |
| 65          | 3.938026094934029  | 6.6209690064319044 |

#### Coefficient of Variation

| Temperature | Energy Inc.        | Time Inc.          |
|:-----------:|-------------------:|-------------------:|
| 45          | -3.0449100419459483 | -6.97293389488448 | 
| 55          | 1.1723675321125389 | -1.309311806802708 |
| 65          | 3.195480698317402 | -5.800776146403544 |